lambdas/thumbnail/tests/test_thumbnail.py

```diff
@@ -144,7 +146,19 @@ def test_generate_thumbnail(
     # Create the lambda request event
     event = _make_event({"url": url, **params})
     # Get the response
-    response = t4_lambda_thumbnail.lambda_handler(event, None)
+    if expected_thumb == "I16-mode-128-fallback.png":
```

Comment:
I don't like piggybacking parameter meanings -- but in this case, it might be the least disruptive thing.  When we're testing for the result png `I16-mode-128-fallback.png`, we do the fallback test.

The only alternative I see is to add a new parameter to the test, `True` for the fallback test, and `False` for every other test.  It seems a bit overkill, which is why I went the way I did.  I don't like either option, tbh, and am open to either -- or any ideas you have.

Reply:
This feels wrong. Can't we just have different image format string to hit the fallback case? I'd rather see us patch something out in the tested function than add a test-only boolean like this.

Reply:
Yeah, I don't like it either.. ..but..

The image mode string is generated when the image is read in (or created from an array).  Just changing the mode doesn't change the bytes, it just changes how PIL treats them.  We would still need valid original and thumbnail images to test against.  PIL/PILLOW has a [plethora of testing images](https://github.com/python-pillow/Pillow/tree/main/Tests/images), as you may imagine -- and I'm sure an image of each mode is accounted for in there, so we could use those (if the licensing is OK) for testing exhaustively -- but it'd be time consuming.

I would ideally like to keep all testing-related code separate from the actual code -- but the basic issue is that if there's a mode that we're certain is a failure case, we may as well fix it and match the quality of our existing thumbnails.  But then that mode is no longer a failure case, and the fallback wouldn't be triggered. 

The current fix fixes two things:
* The mode "I;16", which doesn't have the normal scaling filters, now generates a normal-quality thumbnail, where it didn't generate any thumbnail before
* any undiscovered mode which doesn't have the normal scaling filters will still generate a thumbnail, so long as that mode has a scaling filter of any quality.  The bool is to test this segment of code.

Some alternate options I can think of:
* Ditch the global bool, and use a parameter like `fallback_only=False` to the `generate_thumbnail` function, and call it with `True` when testing the fallback.  Substantially similar to what we're doing now, minus the global.
* remove the bool, and either
  * remove the fallback handling, and just let any new failures fail to generate a thumbnail, and handle each failing mode when/if it happens, resulting in some potential missing thumbnails, but a known fix if they pop up
  * or switch to fallback-only on failure of the normal scaling, so that the thumbnail quality on `I;16` mode is reduced, but the code path is tested by the current tests and needs no bool for testing (neither as a parameter nor as global)
  * or just don't test that segment of code (but that seems wrong, too)